### PR TITLE
mod: update pd client

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/47421 ref https://github.com/tikv/pd/issues/8349

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

File: pkg/executor/internal/calibrateresource/calibrate_resource_test.go [hunk 1]
```diff
@@ -54,17 +54,15 @@ func TestCalibrateResource(t *testing.T) {
 	defer func() {
 		do.SetResourceGroupsController(oldResourceCtl)
 	}()
-
+	// changed in 7.5, but for test pass, use the old config
```
Comment (JmPotato) [inline pkg/executor/internal/calibrateresource/calibrate_resource_test.go:57]: Better attach the change's commit hash.

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**